mrmustard/physics/representations.py

```diff
@@ -373,10 +374,30 @@ def __getitem__(self, idx: int | tuple[int, ...]) -> Bargmann:
         new._contract_idxs = idx
         return new
 
-    def __matmul__(self, other: Bargmann) -> Bargmann:
+    def __matmul__(self, other: Union[Bargmann, Fock]) -> Union[Bargmann, Fock]:
         r"""
         The inner product of ansatze across the marked indices.
+
+        If ``other`` is ``Fock``, then ``self`` is converted to ``Fock`` before the contraction.
+
+        Args:
+            other: Another representation.
+
+        Returns:
+            A ``Bargmann`` representation if ``other`` is ``Bargmann``, or a ``Fock``representation
+            if ``other`` is ``Fock``.
         """
+        idx_z = self._contract_idxs
+        idx_zconj = other._contract_idxs
+
+        if isinstance(other, Fock):
+            from .converters import to_fock
+
+            shape = [settings.AUTOCUTOFF_MAX_CUTOFF for _ in range(len(self.b[0]))]
```

Comment:
```suggestion
            shape = [settings.AUTOCUTOFF_MAX_CUTOFF for _ in range(len(self.b[-1]))]
```

Reply:
It's the same. `self.b` is batched, so `self.b[0]` and `self.b[-1]` both select one batch

Reply:
@ziofil `len(self.b[0])` comes up pretty often, and seems to have the same meaning as `len(self.array)` for Fock. It'd be handy to have a property  ("batch_size"?) for this concept, and it would increase readability. Would you be against it?

Reply:
haha, I had misread! I thought I had read `range(self.b.shape[0])` and I thought I wrote `range(self.b.shape[0])` 😂